### PR TITLE
Hack-#7

### PR DESCRIPTION
**Pull request by brianndiiri**
Customize Favicon for a specific page.

### CODE REVIEW

File: webflow-hacks/misc-hacks/7.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  and store it in the variable link*/
+  const link = document.querySelector<HTMLAnchorElement>("link[rel*='icon']") || document.createElement('link');
+  // set the link type attribute 
+  link.type = 'image/x-icon';
+  // set the link rel attribute
+  link.rel = 'shortcut icon';
+  // set the link href attribute
+  link.href = 'https://assets.website-files.com/5b8f3bd5334d376d704155c1/5c3610ef8b9a3107c28c79ae_hack7-custom-fav.png';
+  // append the link to the head element
+  document.getElementsByTagName('head')[0].appendChild(link);
```
Comment (alexiglesias93) [inline webflow-hacks/misc-hacks/7.ts]: There's a simpler approach:
```typescript
  document.head.appendChild(link);
```

File: webflow-hacks/misc-hacks/7.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// when the DOM is ready
+document.addEventListener('DOMContentLoaded', function () {
+  /* get a reference to the link element
+  with a rel attribute of icon and store it in the variable link or 
+  if the link does not exist, create a link element
+  and store it in the variable link*/
+  // add selector constant
+  const LINK_SELECTOR = "[fs-hacks-element='link-icon']";
+  const link = document.querySelector<HTMLAnchorElement>(LINK_SELECTOR) || document.createElement('link');
```
Comment (alexiglesias93) [inline webflow-hacks/misc-hacks/7.ts]: I think this should not be a custom selector querying a random anchor link on the page.
I believe the orignal script queries the `<link>` element that holds the favicon.

File: webflow-hacks/misc-hacks/7.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// when the DOM is ready
+document.addEventListener('DOMContentLoaded', function () {
+  const FAVICON_URL =
+    'https://assets.website-files.com/5b8f3bd5334d376d704155c1/5c3610ef8b9a3107c28c79ae_hack7-custom-fav.png';
+  const LINK_SELECTOR = "link[rel*='icon']";
+  const link = document.querySelector<HTMLAnchorElement>(LINK_SELECTOR) || document.createElement('link');
```
Comment (alexiglesias93) [inline webflow-hacks/misc-hacks/7.ts:6]:
```typescript
const link = document.querySelector<HTMLLinkElement>("link[rel*='icon']") || document.createElement('link');
```

**Review by alexiglesias93 — CHANGES_REQUESTED** [review]

File: webflow-hacks/misc-hacks/7.ts (hunk 3 above)
Comment (alexiglesias93) [inline webflow-hacks/misc-hacks/7.ts]:
```diff
- const link = document.querySelector<HTMLAnchorElement>(LINK_SELECTOR) || document.createElement('link');
+ const link = document.querySelector<HTMLLinkElement>(LINK_SELECTOR) || document.createElement('link');
```

**Review by alexiglesias93 — CHANGES_REQUESTED** [review]

**Comment by alexiglesias93** [comment]
@brayanfinsweet can you check the merging conflicts? Why does this branch remove the `webflow-hacks/form-hacks/1.ts` files?

**Comment by brianndiiri** [comment]
Hi I will add them back. I wanted to remove them to put inside a folder but I used the wrong branch

**Review by alexiglesias93 — APPROVED** [review]

**Pull request merged**